spec/services/declarations/create_spec.rb

```diff
@@ -0,0 +1,216 @@
+  let(:delivery_partner) { training_period.delivery_partner }
+  let(:payment_statement) { FactoryBot.create(:statement, :open, active_lead_provider:) }
+  let(:existing_duplicate_declaration) { nil }
+
+  let(:service) do
+    described_class.new(
+      author:,
+      lead_provider:,
+      teacher:,
+      training_period:,
+      declaration_date:,
+      declaration_type:,
+      evidence_type:,
+      payment_statement:,
+      mentorship_period:,
+      delivery_partner:,
+      existing_duplicate_declaration:
+    )
+  end
+
+  describe "#create" do
+    subject(:create_declaration) { service.create }
+
+    %i[ect mentor].each do |trainee_type|
+      context "for #{trainee_type}" do
+        let(:teacher_type) { trainee_type }
+
+        context "when there's no existing declaration" do
+          let(:at_school_period) { FactoryBot.create(:"#{trainee_type}_at_school_period", started_on: 6.months.ago, finished_on: 2.weeks.from_now) }
+          let!(:training_period) { FactoryBot.create(:training_period, :"for_#{trainee_type}", :active, "#{trainee_type}_at_school_period": at_school_period, started_on: at_school_period.started_on, finished_on: at_school_period.finished_on) }
+          let!(:mentorship_period) do
+            if trainee_type == :ect
+              mentor = FactoryBot.create(:mentor_at_school_period, started_on: at_school_period.started_on, finished_on: at_school_period.finished_on)
+              FactoryBot.create(:mentorship_period, mentee: at_school_period, mentor:, started_on: at_school_period.started_on, finished_on: at_school_period.finished_on)
+            end
+          end
+
+          it "creates a new declaration with correct attributes" do
+            declaration = nil
+            expect { declaration = create_declaration }.to change(Declaration, :count).by(1)
+
+            expect(declaration.payment_statement).to be_nil
+            if trainee_type == :ect
+              expect(declaration.mentorship_period).to eq(mentorship_period)
+            end
+            expect(declaration.evidence_type).to eq(evidence_type)
+            expect(declaration.payment_status_eligible?).to be(false)
+          end
+
+          if trainee_type == :mentor
+            it "runs mentor completion" do
+              allow(Declarations::MentorCompletion).to receive(:new).and_call_original
+
+              declaration = create_declaration
+
+              expect(Declarations::MentorCompletion).to have_received(:new).with(author:, declaration:).once
+            end
+          end
+
+          context "when teacher is eligible for funding" do
+            before do
+              teacher.update!("#{trainee_type}_first_became_eligible_for_training_at": 1.year.ago)
+            end
+
+            it "marks declaration `payment_status` as eligible" do
+              declaration = create_declaration
+
+              expect(declaration.payment_status_eligible?).to be(true)
```

Comment:
Just a suggestion but all of these could be changed to this format, which is more readable IMO:

`expect(declaration).to be_payment_status_eligible`